Pomodoro.Api/Controllers/SettingsController.cs

```diff
@@ -0,0 +1,141 @@
+        }
+
+        /// <summary>
+        /// Gets pomodoro settings related to the current user.
+        /// </summary>
+        /// <returns>A <see cref="SettingsViewModel"/> object.</returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [SwaggerResponse(200, "Returns an object containing user settings.")]
+        [SwaggerResponse(401, "An unauthorized request cannot be processed.")]
+        [SwaggerResponse(404, "No settings found for the current user.")]
+        [SwaggerResponse(500, "An unhandled exception occurred on the server while executing the request.")]
+        public async Task<ActionResult<SettingsViewModel>> GetUserSettings()
+        {
+            var settingsModel = await this.settingsService
+                .GetUserSettingsAsync(this.UserId);
+
+            if (settingsModel is null)
+            {
+                return this.NotFound("No settings found for the current user.");
+            }
+
+            var settingsViewModel = this.mapper.Map<SettingsViewModel>(settingsModel);
+            return this.Ok(settingsViewModel);
+        }
+
+        /// <summary>
+        /// Creates pomodoro settings related to the current user.
+        /// </summary>
+        /// <param name="settingsVewModel">Represents object to be created.</param>
+        /// <returns>Created object represented by <see cref="SettingsViewModel"/>.</returns>
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [SwaggerResponse(201, "Returns created object containing user settings.")]
+        [SwaggerResponse(400, "The model state is invalid or there is a mismatch.")]
+        [SwaggerResponse(401, "An unauthorized request cannot be processed.")]
+        [SwaggerResponse(409, "Pomodoro settings for the current user already exist.")]
+        [SwaggerResponse(500, "An unhandled exception occurred on the server while executing the request.")]
+        public async Task<ActionResult<SettingsViewModel>> CreateSettings([FromBody] SettingsViewModel settingsVewModel)
+        {
+            if (this.UserId != settingsVewModel.UserId)
+            {
+                return this.BadRequest("UserId mismatch.");
+            }
+
+            var userSettings = await this.settingsService
+                .GetUserSettingsAsync(this.UserId);
+            if (userSettings != null)
+            {
+                return this.Conflict("Pomodoro settings for the current user already exist.");
```

Comment:
Here trouble in next use case:
I use tracker from another device, where I don't logged in. I change tracker settings, and understand that this setting better this my previous one. So I logged in, and what next? If I recieve settings from server automaticaly, my current (better for me) setting is ovewritted. So owerriting current settings without user confirmation is not a good idea. But in current business logic I can't push to server new settings if don't have on my browser previous one.
Maybe in this case we should delete previous settings and save new?

Reply:
Thanks for the use case. From this perspective, your idea seems better than mine.